src/modules/logicModule/BountyManager.sol

```diff
@@ -0,0 +1,528 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.19;
+
+// External Interfaces
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+
+// External Libraries
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+import {EnumerableSet} from "@oz/utils/structs/EnumerableSet.sol";
+
+// Internal Dependencies
+import {Module} from "src/modules/base/Module.sol";
+
+import {PaymentClient} from "src/modules/base/mixins/PaymentClient.sol";
+
+// Internal Interfaces
+import {IProposal} from "src/proposal/IProposal.sol";
+import {IRoleAuthorizer} from "src/modules/authorizer/IRoleAuthorizer.sol";
+import {IBountyManager} from "src/modules/logicModule/IBountyManager.sol";
+
+import {
+    IPaymentClient,
+    IPaymentProcessor
+} from "src/modules/base/mixins/PaymentClient.sol";
+
+// Internal Libraries
+import {LinkedIdList} from "src/common/LinkedIdList.sol";
+
+//@todo this has a direct dependency to the new RoleAuthorizer Module.
+//At the current state of the project the IRoleAuthorizer isnt the standardized Interface to use,
+//That means there is a possibility function calls will fail if the roleAuthorizer isnt implemented as the Authorizer
+//This will be changed in the coming future,
+//but till then the RoleAuthorizer has to be selected as the Authorizer Module of the proposal if the BountyManager is to be used
+
+contract BountyManager is IBountyManager, Module, PaymentClient {
+    using SafeERC20 for IERC20;
+    using EnumerableSet for EnumerableSet.UintSet;
+    using LinkedIdList for LinkedIdList.List;
+
+    //--------------------------------------------------------------------------
+    // Modifiers
+
+    //@todo Reminder that this will be moved into the Module Contract at a later point of time
+    modifier onlyRole(uint8 roleId) {
+        if (
+            !IRoleAuthorizer(address(__Module_proposal.authorizer()))
+                .isAuthorized(roleId, _msgSender())
+        ) {
+            revert Module__BountyManager__OnlyRole(roleId, address(this));
+        }
+        _;
+    }
+
+    modifier onlyClaimContributor(uint claimId) {
+        address sender = _msgSender();
+        Contributor[] memory contribs = _claimRegistry[claimId].contributors;
+        uint length = contribs.length;
+        bool found;
+        for (uint i; i < length; i++) {
+            if (contribs[i].addr == sender) {
```

Comment:
If we could ensure that the list of contributors is sorted... then we can use binary search instead of linear search, which will reduce the current O(n) complexity to O(log N)

Reply:
If you know an easy implementation that is actually cheaper gas wise than sure we can do that
If you dont I think its probably not a high priority target for optimization, because the functions that use it probably wont be in use that often